Formula/yarn.rb

```diff
@@ -0,0 +1,21 @@
+require "language/node"
+
+class Yarn < Formula
+  desc "Javascript package manager"
+  homepage "https://yarnpkg.com/"
+  url "https://registry.npmjs.org/yarn/-/yarn-0.15.1.tgz"
```

Comment:
This should come from the official Yarn site rather than from npmjs. The download URL is `https://yarnpkg.com/downloads/0.15.0/yarn-v0.15.0.tar.gz`

I submitted a followup PR to adjust the URL (https://github.com/Homebrew/homebrew-core/pull/5878) but unfortunately I'm not on a Mac so I can't test it out.